src/PollingBlockTracker.ts

```diff
@@ -106,9 +120,32 @@ export class PollingBlockTracker
       return this._currentBlock;
     }
     // wait for a new latest block
-    const latestBlock: string = await new Promise((resolve) =>
-      this.once('latest', resolve),
-    );
+    const latestBlock: string = await new Promise((resolve, reject) => {
+      // eslint-disable-next-line prefer-const
+      let onLatestBlockUnavailable: InternalListener;
+      const onLatestBlockAvailable = (value: string | PromiseLike<string>) => {
+        this.#removeInternalListener(onLatestBlockAvailable);
+        this.removeListener('error', onLatestBlockUnavailable);
+        resolve(value);
+      };
+      onLatestBlockUnavailable = () => {
+        // if the block tracker is no longer running, reject
+        // and remove the listeners
+        if (!this._isRunning) {
+          this.#removeInternalListener(onLatestBlockAvailable);
+          this.#removeInternalListener(onLatestBlockUnavailable);
+          this.removeListener('latest', onLatestBlockAvailable);
+          this.removeListener('error', onLatestBlockUnavailable);
+          reject(
+            new Error('Block tracker ended before latest block was available'),
+          );
+        }
+      };
+      this.#addInternalListener(onLatestBlockAvailable);
+      this.#addInternalListener(onLatestBlockUnavailable);
+      this.once('latest', onLatestBlockAvailable);
+      this.on('error', onLatestBlockUnavailable);
```

Comment:
this listener is attached using `.on` instead of `.once` because the block tracker could be stopped after several errors, and we need to catch all of them